### PR TITLE
Use containerD WithHostDevices

### LINKED ISSUE #1316: Use new host device options added in containerd
See https://github.com/containerd/containerd/pull/3718

This helps code deduplication.

### PR DESCRIPTION
**Pull request by Zyqsempai**
Fixes #1316 
Signed-off-by: bpopovschi <zyqsempai@mail.ru>

### CODE REVIEW

File: pkg/server/container_create_unix.go [hunk 1]
```diff
@@ -168,10 +168,7 @@ func (c *criService) containerSpec(id string, sandboxID string, sandboxPid uint3
 		if !sandboxConfig.GetLinux().GetSecurityContext().GetPrivileged() {
 			return nil, errors.New("no privileged container allowed in sandbox")
 		}
-		specOpts = append(specOpts, oci.WithPrivileged)
-		if !ociRuntime.PrivilegedWithoutHostDevices {
```
Comment (mikebrow) [inline pkg/server/container_create_unix.go]: could we add some comments here... I don't follow why are we removing/ignoring the privilegedwithouthostdevices config directive and replacing it with always withhostdevices when privileged.  

	// PrivilegedWithoutHostDevices overloads the default behaviour for adding host devices to the
	// runtime spec when the container is privileged. Defaults to false.

File: pkg/server/container_create_unix.go [hunk 2]
```diff
@@ -168,10 +168,7 @@ func (c *criService) containerSpec(id string, sandboxID string, sandboxPid uint3
...
-			specOpts = append(specOpts, customopts.WithPrivilegedDevices)
```
Comment (mikebrow) [inline pkg/server/container_create_unix.go]: I suspect you should just replace WithPrivilegedDevices with WithHostDevices. @Random-Liu thoughts?

Comment (Random-Liu) [reply pkg/server/container_create_unix.go]: Yeah, and remove the code for `WithPrivilegedDevices`.

**Comment by Random-Liu** [comment]
It seems that this change broke the kubernetes test.

The change was merged at 12/5 2:13 PM
The next build was at 12/5 2:15 PM
The first test failure started at 12/5 2:20 PM

https://k8s-testgrid.appspot.com/sig-node-containerd#e2e-gci

I'll revert this change to bring the test back to normal, and let's investigate this more.

**Pull request merged**